tutor/src/screens/my-courses/dashboard/index.tsx

```diff
@@ -109,54 +111,45 @@ const StyledMyCoursesDashboard = styled.div`
  */
 
 export const MyCoursesDashboard = () => {
+    const dispatch = useDispatch()
     // getting all the data: offerings and courses
     const courses = useAllCourses()
     const offerings = useAvailableOfferings(courses)
 
     const [deleteOfferingIdModal, setDeleteOfferingIdModal] = useState<ID | null>(null)
-    const [displayedOfferingIds, setDisplayedOfferingIds] = useState<ID[]>(
-        useDisplayedCourseOfferingIds()
-    )
+    const [displayedOfferingIds, setDisplayedOfferingIds, displayedOfferings, swapOffering] = useDisplayedOfferings()
     const [isEditMode, setIsEditMode] = useState<boolean>(false)
 
-    // update the `displayedOfferingIds` if users adds/delete offerings
-    useEffect(() => {
-        UiSettings.set('displayedOfferingIds', [...displayedOfferingIds])
-    }, [displayedOfferingIds])
-
-    // re-compute the `displayedOfferings` if `displayedOfferingIds` is modified
-    const displayedOfferings = useMemo(() => {
-        const filteredOfferings = filter(offerings, o => includes(displayedOfferingIds, o.id))
-        return filteredOfferings.sort((a, b) => {
-            return indexOf(displayedOfferingIds, a.id) - indexOf(displayedOfferingIds, b.id)
-        })
-    }, [displayedOfferingIds, offerings])
-
-    // move offerings block around
-    const swapOffering = (offeringId: ID, flow = 'up') => {
-        const tempDisplayedOfferingIds = [...displayedOfferingIds]
-        const index = findIndex(tempDisplayedOfferingIds, id => id === offeringId)
-        if (index >= 0) {
-            const swapIndex = flow === 'up' ? index - 1 : index + 1;
-            [tempDisplayedOfferingIds[index], tempDisplayedOfferingIds[swapIndex]] = [tempDisplayedOfferingIds[swapIndex], tempDisplayedOfferingIds[index]];
-        }
-        setDisplayedOfferingIds(tempDisplayedOfferingIds)
-    }
-
     // delete offering
     const deleteOffering = () => {
         const tempDisplayedOfferingIds = [...displayedOfferingIds]
         const index = findIndex(tempDisplayedOfferingIds, id => id === deleteOfferingIdModal)
-        if (index >= 0) {
-            tempDisplayedOfferingIds.splice(index, 1)
-            setDisplayedOfferingIds(tempDisplayedOfferingIds)
-            setDeleteOfferingIdModal(null)
+        if (index >= 0 && deleteOfferingIdModal) {
+            const offeringCourses = filter(courses, c => c.offering_id === deleteOfferingIdModal && String(c.term) !== 'preview')
+            Promise.all(map(offeringCourses, c => {
+                const currentRole = find(c.roles, r => r.type === 'teacher')
+                const currentTeacher = currentRole && find(c.teachers, t => t.role_id === currentRole.id);
+                if(currentTeacher) {
+                    dispatch(dropCourseTeacher(currentTeacher.id))
+                } else {
+                    Promise.resolve()
+                }
+            }))
```

Comment:
I think you need to return a promise here, either the result of dispatch or Promise.resolve.  

since the function that is iterating through map never returns anything you're ending up with `[undefined]` and promise.all won't wait on anything